core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -4004,6 +4471,99 @@ class KafkaApis(val requestChannel: RequestChannel,
     CompletableFuture.completedFuture[Unit](())
   }
 
+
+  def getAcknowledgeBatchesFromShareFetchRequest(
+                                                  shareFetchRequest : ShareFetchRequest,
+                                                  topicNames : util.Map[Uuid, String],
+                                                  erroneous : mutable.Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData],
+                                                ) : mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]] = {
+
+    val acknowledgeBatchesMap = mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]]()
+    shareFetchRequest.data().topics().forEach ( topic => {
+
+      if(!topicNames.asScala.contains(topic.topicId)) {
+        topic.partitions.forEach((partition: ShareFetchRequestData.FetchPartition) => {
+          val topicIdPartition = new TopicIdPartition(
+            topic.topicId,
+            new TopicPartition(null, partition.partitionIndex))
+          erroneous +=
+            topicIdPartition -> ShareAcknowledgeResponse.partitionResponse(topicIdPartition, Errors.UNKNOWN_TOPIC_ID)
+        })
+      }
+      else {
+        topic.partitions().forEach ( partition => {
+          val topicIdPartition = new TopicIdPartition(
+            topic.topicId(),
+            new TopicPartition(topicNames.get(topic.topicId()), partition.partitionIndex())
+          )
```

Comment:
Do we validate somewhere that partition index exists for the topic i.e. what if client request for partition 5 when there exists only 4 partitions for topic?

Reply:
Not currently. I think that can be done using the MetadataCache which is accessible in the KafkaApis. I will create a separate JIRA for this. Thanks !

Reply:
Do we have the jira now? Can we please link here?

Reply:
Hey yep, the JIRA has been created - https://issues.apache.org/jira/browse/KAFKA-17127